clients/src/main/java/org/apache/kafka/clients/admin/TopicDescription.java

```diff
@@ -65,6 +65,19 @@ public TopicDescription(String name, boolean internal, List<TopicPartitionInfo>
         this(name, internal, partitions, Collections.emptySet());
     }
 
+    /**
+     * Create an instance with the specified parameters.
+     *
+     * @param name The topic name
+     * @param internal Whether the topic is internal to Kafka
+     * @param partitions A list of partitions where the index represents the partition id and the element contains
+     *                   leadership and replica information for that partition.
+     * @param topicId the topic id
+     */
+    public TopicDescription(String name, boolean internal, List<TopicPartitionInfo> partitions, Uuid topicId) {
```

Comment:
Do we really need this constructor in the public class? We could just use the one below?